### ISSUE TITLE
[Android] The app is intermittently not installed the first time and can not be re-installed on subsequent tries

### ISSUE THREAD

**Issue by Softvision-AndreeaCupsa** [mozilla/send#1234]
**[Notes]:**
- On some devices, I managed to install the application on the first try (Huawei P20 Lite, Google Pixel, Oneplus 5) but on Samsung S8 I was not able to do so and the “App not installed” message was displayed. After uninstalling the application and attempting to reinstall it, the “App not installed” message was displayed on all our testing devices (and personal).
- Deleting the cache and data of the “Google Play Store” application made it possible to reinstall the application.
- On Google Pixel the “App not installed” message was displayed on my first attempt to reinstall the application, but I was able to install it on a second try without deleting the cache and data of the “Google Play Store” application.

**[Affected Platforms]:**
- Android 8 (Samsung S8, Huawei P20 Lite, Google Pixel)
- Android 9 (Oneplus 5) 

**[Prerequisites]:**
- Have the latest Send Android application (org.mozilla.firefoxsend-2019-03-12-beta-release-signed) save on the device.

**[Steps to reproduce]:**
1. Navigate to the folder where the Send apk is saved.
2. Tap the org.mozilla.firefoxsend-2019-03-12-beta-release-signed.apk file.
3. Tap the “Install” button.
4. Tap the “Install anyway” button (if prompted) and observe the behavior.

**[Expected result]:**
- The app is successfully installed.

**[Actual result]:**
- The “App not installed” message is displayed. 

**[Notes]:**
- Attached a screen recording with the issue: [Link](https://drive.google.com/open?id=1MAkgllwp-Cl05gq_2zSAO418RICP1VW0)

**Comment by johngruen** [comment]
@Softvision-AndreeaCupsa is there any logging beyond this?

**Comment by Softvision-AndreeaCupsa** [comment]
Sure.
I attached a file with all logs and one with only the error logs:
- [log_err.txt](https://github.com/mozilla/send/files/2962265/log_err.txt)
- [log_all.txt](https://github.com/mozilla/send/files/2962266/log_all.txt)

**Comment by g-k** [comment]
@johanlorenzo have you seen this before for our other APKs?

**Comment by JohanLorenzo** [comment]
This is the first time I see this behavior. The logs seem to point to that problem https://stackoverflow.com/questions/51310155/installation-silently-blocked-for-android-release-build-signed-apk-when-testing, though: 

```
03-13 17:38:48.151  9587  9698 I Finsky  : [496] twj.b(3): Verification id=12 response=POTENTIALLY_UNWANTED
03-13 17:38:48.158  3783  4243 D PackageManager: [VERIFY] extendVerificationTimeout(12, REJECT, 3600000), uid=10054
03-13 17:38:48.158  3783  4243 D PackageManager: [VERIFY] extendVerificationTimeout: timeout extended
03-13 17:38:48.158  3783  4243 D PackageManager: [VERIFY] extendTimeout(12), uid=10054
03-13 17:38:48.176  3237  3237 I android.hardware.wifi@1.0-service: getLinkLayerStats
03-13 17:38:48.241  9587  9734 I Finsky  : [511] twj.b(3): Verification id=12 response=POTENTIALLY_UNWANTED
03-13 17:38:48.261  9587  9698 I Finsky  : [496] txc.c(3): Verifying id=12, result=-1
03-13 17:38:48.262  3783  4243 D PackageManager: [VERIFY] verifyPendingInstall(12, REJECT), uid=10054
03-13 17:38:48.263  3783  3823 D PackageManager: [VERIFY] PACKAGE_VERIFIED: 
03-13 17:38:48.263  3783  3823 D PackageManager: PackageVerificationState{
03-13 17:38:48.263  3783  3823 D PackageManager:     verifier packages=com.android.vending(10054),com.samsung.android.sm.devicesecurity(5012),
03-13 17:38:48.263  3783  3823 D PackageManager:     intent=Intent { act=android.intent.action.PACKAGE_NEEDS_VERIFICATION dat=file:///data/app/vmdl1976235347.tmp typ=application/vnd.android.package-archive flg=0x10000001 cmp=com.android.vending/com.google.android.finsky.verifier.impl.PackageVerificationReceiver (has extras) }
03-13 17:38:48.263  3783  3823 D PackageManager:     others=12, false, false, false, false, true
03-13 17:38:48.263  3783  3823 D PackageManager: }
03-13 17:38:48.263  3783  3823 D PackageManager: [VERIFY] setVerifierResponse {verificationID=12, uid=10054, code=-1}
03-13 17:38:48.263  3783  3823 D PackageManager: [VERIFY] packageVerificationCompletion: completed id=12
03-13 17:38:48.274  3783  3823 D PackageManager: result of install: -22{123425927}
03-13 17:38:48.278  9587  9698 I Finsky  : [496] twb.a(19): Installation silently blocked for org.mozilla.firefoxsend
03-13 17:38:48.279  9587  9587 I Finsky  : [2] txc.b(6): Verification complete: id=12, package_name=org.mozilla.firefoxsend
```


Per the STRs, the problem occurs when installing the APK manually. Does this also happen when installing the app from Google Play, directly, @Softvision-AndreeaCupsa ?

**Comment by Softvision-AndreeaCupsa** [comment]
@JohanLorenzo The application was not posted yet on the Google Play Store, so I can not test this situation.

**Comment by JohanLorenzo** [comment]
Oh you're right. I saw the app was marked as published on the Google Play console, but it's just a dummy APK there. Sorry for the confusion!

**Comment by johngruen** [comment]
Plan is we're going to do an internal Play Store release and test

**Comment by johngruen** [comment]
@JohanLorenzo @Softvision-AndreeaCupsa confirms that this issue is not reproducible in GPS.

**Comment by JohanLorenzo** [comment]
I see. I guess `Finsky` blocked the APK because it was the first time Google saw it. I guess things will go better from now on (now that the APK is on Google Play).

@Softvision-AndreeaCupsa do you still reproduce with your initial STRs?

**Comment by johngruen** [comment]
@JohanLorenzo what is Finsky?

**Comment by JohanLorenzo** [comment]
Finsky (called out in the above logs) is the name of one component of the Google Play Store https://play.google.com/store/apps/details?id=com.google.android.finsky. I don't know too much about this component. I just saw it blocked the installation of the APK.

**Comment by Softvision-AndreeaCupsa** [comment]
@JohanLorenzo No, I don't reproduce the issue with the initial build (org.mozilla.firefoxsend-2019-03-12-beta-release-signed).
